From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Sun, 27 Oct 2024 20:53:14 +0100
Subject: [PATCH] Fix double mobspawn counting

Mentioned in #paper-contrib channel on PaperMC's discord, but they didn't fix it on 1.21.1

See https://canary.discord.com/channels/289587909051416579/925530366192779286/1300043236827594882

diff --git a/src/main/java/net/minecraft/world/level/NaturalSpawner.java b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
index ae4c446ca19408ead11a1d775d3042c43b019699..d433058ad15dbcab028401e2efa99da8f0312371 100644
--- a/src/main/java/net/minecraft/world/level/NaturalSpawner.java
+++ b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
@@ -578,7 +578,7 @@ public final class NaturalSpawner {
             this.spawnPotential.addCharge(blockposition, d0);
             MobCategory enumcreaturetype = entitytypes.getCategory();
 
-            this.mobCategoryCounts.addTo(enumcreaturetype, 1);
+            // this.mobCategoryCounts.addTo(enumcreaturetype, 1); // CloudPlane - fix double mobspawn counting
             if (this.localMobCapCalculator != null) this.localMobCapCalculator.addMob(new ChunkPos(blockposition), enumcreaturetype); // Paper - Optional per player mob spawns
         }
 
